### PR TITLE
string formatting lab complete

### PR DESCRIPTION
**Pull request by natrod22**

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
@natrod22 hi there, I'm assuming your mailroom is not ready for review?

**Comment by natasha-aleksandrova** [comment]
str format lab is good to merge

**Comment by robalford** [comment]
@natasha-aleksandrova  Natalie would like to close this out and open a new PR for her mailroom work (we've been in touch over email)

**Comment by natrod22** [comment]
mailroom_final.py is ready for review

File: students/NatalieRodriguez/Lesson03/mailroom_final.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          ['River Tails', 63.56, 1200.00, 300.65], ['Virginia Ferdinand', 350.35, 5000.00], ['Joseph Kibson', 3498.00, 5.50],]
+
+
+def thank_you():
+    print("\nPlease enter the name of the donor you would like to thank,\nenter a new name, or enter 'list' for current donors.\nEnter '3' to return to the MAILROOM MENU.")
+    answer = input()
+
+    if answer == "list":
+        for donor_total in donors:
+            print(donor_total[0])
```
Comment (natasha-aleksandrova) [inline students/NatalieRodriguez/Lesson03/mailroom_final.py]: readability tip: this should be `for donor in donors` as you are not iterating over totals but donor entry

File: students/NatalieRodriguez/Lesson03/mailroom_final.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        return
+    else:
+        print("Enter the amount donated:")
+        amount = input()
+
+        cur = [answer, float(amount)]
+        for donor_total in donors:
+            if cur[0] == donor_total[0]:
+                donor_total.append(cur[1])
+                break
```
Comment (natasha-aleksandrova) [inline students/NatalieRodriguez/Lesson03/mailroom_final.py]: above is a bit confusing, consider below:
```
for donor_name, donations in donors:
    if answer == donor_name:
        donations.append(amount)
        break
```

File: students/NatalieRodriguez/Lesson03/mailroom_final.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        else:
+            donors.append(cur)
```
Comment (natasha-aleksandrova) [inline students/NatalieRodriguez/Lesson03/mailroom_final.py]: nice job on adding the else clause

File: students/NatalieRodriguez/Lesson03/mailroom_final.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    
+    
+    donor_list = list()
+    for donor_data in donors:
+        gift_amount = [donor_data[0], 0, 0, 0]
+        for donation in donor_data[1:]:
+            gift_amount[1] += donation
+            gift_amount[2] += 1
+        gift_amount[3] = gift_amount[1] / gift_amount[2]
+        donor_list.append(gift_amount)
```
Comment (natasha-aleksandrova) [inline students/NatalieRodriguez/Lesson03/mailroom_final.py]: your other mailroom version was using sum and len which is preferable (choose readable built-in functions over doing something manually):
```
for donor_name, donation in donors:
    donor_list.append([donor_name, sum(donation), len(donation), sum(donation) / len(donation)])
```

File: students/NatalieRodriguez/Lesson03/mailroom_final.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    # Sort the list by total amount given
+    donor_list.sort(key=get_total, reverse=True)
```
Comment (natasha-aleksandrova) [inline students/NatalieRodriguez/Lesson03/mailroom_final.py]: note that there is a utility function that is really helpful in sorting (especially when you're sorting list of dictionaries)
```
from operator import itemgetter
donor_list.sort(key=itemgetter(1), reverse=True)
```

**Pull request merged**